### DISCUSSION TITLE
HikariCp-3.4.5 multi-threaded support issue

### DISCUSSION THREAD

**Discussion by zjj1002** [apache/skywalking#11676] [Q&A]
HI Community:

When conn pooling is not used in my code, I can see all SPANs and trace links are normal.

But when I used HikariCp-3.4.5, I found that only 1/4 of the requests had a complete span, and many trace links were disconnected from the MYSQL query method. 

A new tracking ID was automatically inserted from the MYSQL query.

The tracking ID I generated myself from the source of the request is broken here, replaced by the automatically generated traceID.

May I ask if this is caused by incomplete multithreading support?

If so, how can we avoid or prevent it?

Thank you !!!

**Comment by wu-sheng** [comment]
You need to see where the trace ID is from. Is somehow this connection pool reuse statement, or something?

This scenario doesn't seem to multiple thread, instead, more like an object reusing.

**Comment by zjj1002** [comment]
You are right, it is indeed a problem of link reuse, but how can I handle this situation?

**Reply by wu-sheng** [reply]
I don't know, you need to debug codes to see how this gets cached and causes dirty data.

**Comment by zjj1002** [comment]
Thank you!
I think this is a very serious bug, which basically leads to the unavailability of Hikari. Have the community encountered similar problems?

**Reply by wu-sheng** [reply]
Community is a very general and abstract idea. There is no physical individuals to connect.
If you are in urgent, you should take actions to debug. If not, this is where people may read.
